src/configuration.cpp

```
@@ -161,74 +160,154 @@ chimera::CompiledConfiguration::CompiledConfiguration(
         namespaces_.insert(ns);
     }
 
-    // Resolve namespace configuration entries within provided AST.
-    for(const auto &it : configNode["namespaces"])
+    // Get a reference to the configuration YAML root node.
+    // If it is not available, then skip the remaining parsing.
+    if (!configNode_)
+        return;
+
+    // Parse 'namespaces' section of configuration YAML if it exists.
+    const YAML::Node &namespacesNode = configNode_["namespaces"];
+    if (namespacesNode)
     {
-        std::string ns_str = it.first.as<std::string>();
-        auto ns = chimera::util::resolveNamespace(ci, ns_str);
-        if (ns)
+        // Check that 'namespaces' node in configuration YAML is a map.
+        if (!namespacesNode.IsMap())
         {
-            declarations_[ns] = it.second;
-            namespaces_.insert(ns);
+            std::cerr << "'namespaces' in configuration YAML must be a map."
+                      << std::endl;
+            exit(-2);
```

Can we throw an exception instead of `exit`ing here? 😭